api/models.py

```diff
@@ -1,18 +1,108 @@
+import uuid
+
+from django.contrib.auth.models import AbstractBaseUser, BaseUserManager
 from django.db import models
-from django.contrib.auth.models import User
+
+
+class UserManager(BaseUserManager):
+    """
+    creating a manager for a custom user model
+    https://docs.djangoproject.com/en/3.0/topics/auth/customizing/#writing-a-manager-for-a-custom-user-model
+    https://docs.djangoproject.com/en/3.0/topics/auth/customizing/#a-full-example
+    """
+
+    def create_user(self, email, password=None):
+        """
+        Create and return a `User` with an email, username and password.
+        """
+        if not email:
+            raise ValueError("Users Must Have an email address")
+
+        user = self.model(
+            email=self.normalize_email(email),
+        )
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_superuser(self, email, password):
+        """
+        Create and return a `User` with superuser (admin) permissions.
+        """
+        if password is None:
+            raise TypeError("Superusers must have a password.")
+
+        user = self.create_user(email, password)
+        user.is_superuser = True
+        user.is_staff = True
+        user.save()
+
+        return user
+
+
+class User(AbstractBaseUser):
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    email = models.EmailField(verbose_name="email address", max_length=255, unique=True)
+    is_active = models.BooleanField(default=True)
+    is_staff = models.BooleanField(default=False)
+    is_superuser = models.BooleanField(default=False)
+    USERNAME_FIELD = "email"
+    REQUIRED_FIELDS = []
+
+    # Tells Django that the UserManager class defined above should manage
+    # objects of this type.
+    objects = UserManager()
+
+    def __str__(self):
+        return self.email
+
+    class Meta:
+        """
+        to set table name in database
+        """
+
+        db_table = "login"
+
+
+# Note that enforcement on these fields should be done on the frontend --- Very easy!
+class UserProfile(models.Model):
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    user = models.OneToOneField(User, on_delete=models.CASCADE, related_name="profile")
+    first_name = models.CharField(max_length=50, unique=False)
+    last_name = models.CharField(max_length=50, unique=False)
+    phone_number = models.CharField(max_length=10, unique=True, null=True)
+    age = models.PositiveIntegerField(null=True, blank=False)
+    GENDER_CHOICES = (
+        ("M", "Male"),
+        ("F", "Female"),
+    )
+    gender = models.CharField(max_length=1, choices=GENDER_CHOICES, null=True)
+
+    class Meta:
+        """
+        to set table name in database
```

Comment:
I'd leave this out because it is a bit redundant and not entirely accurate. The `Meta` class can do more than specify the table name.